packages/cli/src/lib/container/docker_helper.ts

```diff
@@ -163,9 +163,19 @@ export const chooseJobs = async function (jobs: groupedDockerJobs) {
   return chosenJobs
 }
 
-export const buildImage = async function (dockerfile: string, resource: string, buildArgs: string[], path?: string) {
+type BuildImageParams = {
+  dockerfile: string,
+  resource: string,
+  buildArgs: string[],
+  path?: string,
+  arch?: string,
+}
+
+export const buildImage = async function ({dockerfile, resource, buildArgs, path, arch}: BuildImageParams) {
   const cwd = path || Path.dirname(dockerfile)
-  const args = ['build', '-f', dockerfile, '-t', resource, '--platform', 'linux/amd64']
+  const args = ['build', '-f', dockerfile, '-t', resource]
+  // allows for pushing docker build from m1/m2 Macs
+  if (arch === 'arm64') args.push('--platform', 'linux/amd64')
```

Comment:
My understanding from https://github.com/heroku/cli/issues/2967 is that the problem arises from using an old version of the Docker engine, not necessarily because of the architecture. FWIW, that customer is running a version of the Docker engine that is no longer supported -- https://endoflife.date/docker-engine

I'm ok with this solution I guess, because it works for right now and helps that one customer, though I sorta think a "use the platform flag when the version of docker supports it" is probably more what we want.

Reply:
I understand that. I looked into that option but decided to go this direction because we really only need the `--platform` flag for m1/m2 macs. So, for example, someone using a Windows machine or an intel mac doesn't need this flag and they may prefer to not have it.

Reply:
👍 yeah i'm good with shipping as is. it works. maybe it's worth adding a comment to say something like "older docker versions don't support this flag, so we'll only apply this flag in cases where it's definitely broken without this flag."

Reply:
Updated. Thanks!